### PR TITLE
Feature composition

### PR DESCRIPTION
**Pull request by rclapp**
Extract composition from query

### CODE REVIEW

File: mdxpy/mdx.py [hunk 1]
```diff
@@ -1259,6 +1283,29 @@ def __init__(self, cube: str):
...
+            for tuple in self.axes[axis_number].tuples:
+                for hierarchy in tuple.members:
+                    composition.append(hierarchy.hierarchy_unique_name)
+        if self.axes[axis_number].dim_sets:
+            for set in self.axes[axis_number].dim_sets:
+                composition.append(set.hierarchy_unique_name)
+
+        return composition
+
+    def get_composition(self):
```
Comment (MariusWirtz) [inline mdxpy/mdx.py:1298]: I think it would be nice to have a a few test cases on the MdxBuilder class level

Comment (MariusWirtz) [reply mdxpy/mdx.py:1298]: I think this is a helpful addition. It should be robust, easy to test and maintain.

But the ideal solution for https://github.com/cubewise-code/tm1py/discussions/1124#discussioncomment-9724616 would be a function that can derive the composition based on the raw MDX string.
Ideally, we could create an instance of MdxBuilder from a raw MDX string.

Comment (rclapp) [reply mdxpy/mdx.py:1298]: An MDX lexer and parser, that would be an interesting one.

**Comment by rclapp** [comment]
Added new test cases for the composition methods

**Pull request merged**